content/docs/examples/advanced-gateways/egress-mongo/index.md

```diff
@@ -0,0 +1,1111 @@
+      tcp:
+      - match:
+        - gateways:
+          - istio-egressgateway
+          port: 443
+        route:
+        - destination:
+            host: $MONGODB_HOST
+            port:
+              number: $MONGODB_PORT
+          weight: 100
+    EOF
+    {{< /text >}}
+
+    otherwise:
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: istio-egressgateway
+    spec:
+      selector:
+        istio: egressgateway
+      servers:
+      - port:
+          number: 443
+          name: tls
+          protocol: TLS
+        hosts:
+        - $MONGODB_HOST
+        tls:
+          mode: PASSTHROUGH
+    ---
+    apiVersion: networking.istio.io/v1alpha3
+    kind: DestinationRule
+    metadata:
+      name: egressgateway-for-mongo
+    spec:
+      host: istio-egressgateway.istio-system.svc.cluster.local
+      subsets:
+      - name: mongo
+    ---
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: direct-mongo-through-egress-gateway
+    spec:
+      hosts:
+      - $MONGODB_HOST
+      gateways:
+      - mesh
+      - istio-egressgateway
+      tls:
+      - match:
+        - gateways:
+          - mesh
+          port: $MONGODB_PORT
+          sni_hosts:
+          - $MONGODB_HOST
+        route:
+        - destination:
+            host: istio-egressgateway.istio-system.svc.cluster.local
+            subset: mongo
+            port:
+              number: 443
+      - match:
+        - gateways:
+          - istio-egressgateway
+          port: 443
+          sni_hosts:
+          - $MONGODB_HOST
+        route:
+        - destination:
+            host: $MONGODB_HOST
+            port:
+              number: $MONGODB_PORT
+          weight: 100
+    EOF
+    {{< /text >}}
+
+1.  Refresh the web page of the application again and verify that the ratings are still displayed correctly.
+
+1.  Check the statistics of the egress gateway's Envoy and see a counter that corresponds to your
+    requests to the MongoDB service. If Istio is deployed in the `istio-system` namespace, the command to print the
+    counter is:
+
+    {{< text bash >}}
+    $ kubectl exec -it $(kubectl get pod -l istio=egressgateway -n istio-system -o jsonpath='{.items[0].metadata.name}') -c istio-proxy -n istio-system -- curl -s localhost:15000/stats | grep $MONGODB_PORT | grep ${MONGODB_HOST}.upstream_cx_total
+    cluster.outbound|<your MongoDB port>||<your MongoDB host>.upstream_cx_total: 1
+    {{< /text >}}
+
+#### Cleanup directing TLS Egress traffic through an egress gateway
+
+{{< text bash >}}
+$ kubectl delete serviceentry mongo
+$ kubectl delete gateway istio-egressgateway
+$ kubectl delete virtualservice direct-mongo-through-egress-gateway
+$ kubectl delete destinationrule egressgateway-for-mongo
+{{< /text >}}
+
+### Enable MongoDB TLS egress traffic to arbitrary wildcarded domains
```

Comment:
This part seems like it really could be a separate example of blog.

Reply:
Also, isn't this significant overlap with the wildcard example?

Reply:
@frankbu Please see my summary comment regarding the both issues
https://github.com/istio/istio.io/pull/2347#issuecomment-438935643.